ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadManager.swift

```diff
@@ -525,18 +525,24 @@ public class ResourceDownloadManager {
     }
   }
 
-  internal func resourceUpdateCompletionClosure<Resource: LanguageResource>(for resources: [Resource], handler: CompletionHandler<Resource>?) -> CompletionHandler<Resource> {
+  internal func resourceUpdateCompletionClosure<Resource: LanguageResource>(for resources: [Resource]) -> CompletionHandler<Resource> {
     // Updates should not generate notifications per resource.
     return { package, error in
-      // Do not send notifications for individual resource updates.
-
-      handler?(package, error)
-
-      // After the custom handler operates, ensure that any changes it made are synchronized for use
-      // with the app extension, too.
-      let userDefaults = Storage.active.userDefaults
-      userDefaults.set([Date()], forKey: Key.synchronizeSWKeyboard)
-      userDefaults.synchronize()
+      if let package = package {
+        // Do not send notifications for individual resource updates.
+//        let resourceIDs: [Resource.FullID] = resources.map { $0.typedFullID }
+//        do {
+//          try ResourceFileManager.shared.install(resourcesWithIDs: resourceIDs, from: package)
+//        } catch {
+//          log.error("Error updating resources from package \(package.id)")
+//        }
```

Comment:
These lines were the motivating factor behind this PR; it turns out that the previous typing didn't operate cleanly in this scenario.  The redone typing works far more cleanly here.